src/ophys_etl/modules/suite2p_registration/utils.py

```diff
@@ -3,6 +3,125 @@
+        Name of the dataset to load from the HDF5 file.
+    n_sigma : float
+        Number of standard deviations beyond which a frame is considered an
+        outlier and "empty".
+    logger : callable, optional
+        Function to print warning messages to.
+    """
+    # Load the data.
+    with h5py.File(h5py_name, 'r') as h5_file:
+        frames = h5_file[h5py_key][:]
+    # Find the midpoint of the movie.
+    midpoint = frames.shape[0] // 2
+
+    # We discover empty or extrema frames by comparing the mean of each frames
+    # to the mean of the full movie.
+    means = frames.mean(axis=(1, 2))
+    mean_of_frames = means.mean()
+
+    # Compute a robust standard deviation that is not sensitive to the
+    # outliers we are attempting to find.
+    quart_low, quart_high = np.percentile(means, [25, 75])
+    # Convert the inner quartile range to an estimate of the standard deviation
+    # 0.6745 is the converting factor between the inner quartile and a
+    # traditional standard deviation.
+    std_est = (quart_high - quart_low) / (2 * 0.6745)
+
+    # Get the indexes of the frames that are found to be n_sigma deviating.
+    start_idxs = np.sort(
+        np.argwhere(means[:midpoint] < mean_of_frames - 5 * std_est)).flatten()
+    end_idxs = np.sort(
+        np.argwhere(
+            means[midpoint:] < mean_of_frames - 5 * std_est)).flatten() \
+        + midpoint
+
+    # Get the total number of these frames.
+    lowside = len(start_idxs)
+    highside = len(end_idxs)
+
+    # Check to make sure that the indexes found were only from the start/end
+    # of the movie. If not, throw a warning and reset the number of frames
+    # found to zero.
+    if not np.array_equal(start_idxs,
+                          np.arange(0, lowside, dtype=start_idxs.dtype)):
+        lowside = 0
+        if logger is not None:
+            logger("5 sigma discrepant frames found outside the beginning of "
+                   "the movie. Please inspect the movie for data quality. "
+                   "Not trimming frames from the movie beginning.")
+    if not np.array_equal(end_idxs,
+                          np.arange(frames.shape[0] - highside,
+                                    frames.shape[0],
+                                    dtype=end_idxs.dtype)):
+        highside = 0
+        if logger is not None:
+            logger("5 sigma discrepant frames found outside the end of "
+                   "the movie. Please inspect the movie for data quality. "
+                   "Not trimming frames from the movie end.")
+
+    return (lowside, highside)
+
+
+def reset_frame_shift(frames: np.ndarray,
+                      dy_array: np.ndarray,
+                      dx_array: np.ndarray,
+                      trim_frames_start: int,
+                      trim_frames_end: int):
+    """Reset the frames of a movie and their shifts.
+
+    Shifts the frame back to its original location and resets the shifts for
+    those frames to (0, 0).
+
```

Comment:
Please explicitly mention that you are going to change `frames`, `dx_array` and `dy_array` in-place.